applications/crossbar/src/modules/cb_modules_util.erl

```diff
@@ -75,6 +77,14 @@ range_view_options(Context, MaxRange) ->
         _N -> {CreatedFrom, CreatedTo}
     end.
 
+-spec maybe_clear_outbound_flags(cb_context:context()) -> cb_context:context().
```

Comment:
What is the purpose of this function? Why, when a device fails schema validation, would you need to clear these flags? This is unexpected behaviour and I don't think we can accept this change (and the cb_devices_* changes) unless there's a significant issue we don't realize.

Please enlighten! :)